tools/cli/admin_kafka_commands_test.go

```diff
@@ -0,0 +1,975 @@
+			},
+		},
+		{
+			"filter with workflow and runId",
+			buildFilterFn("my-workflow-id", "run-id-1"),
+			[]*types.ReplicationTask{
+				{HistoryTaskV2Attributes: &types.HistoryTaskV2Attributes{WorkflowID: "my-workflow-id", RunID: "run-id-1"}},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var result []*types.ReplicationTask
+
+			for _, task := range allTasks {
+				if tt.filterFn(task) {
+					result = append(result, task)
+				}
+			}
+
+			assert.Equal(t, tt.expectedFilteredTaskList, result)
+		})
+	}
+}
+
+func TestBuildFilterFnForVisibility(t *testing.T) {
+	allMessages := []*indexer.Message{
+		nil,
+		{},
+		{RunID: ptr.String("run-id-1")},
+		{RunID: ptr.String("run-id-2")},
+		{WorkflowID: ptr.String("my-workflow-id")},
+		{WorkflowID: ptr.String("my-workflow-id"), RunID: ptr.String("run-id-1")},
+		{WorkflowID: ptr.String("my-workflow-id"), RunID: ptr.String("run-id-2")},
+		{WorkflowID: ptr.String("alien-workflow-id")},
+		{WorkflowID: ptr.String("alien-workflow-id"), RunID: ptr.String("run-id-1")},
+		{WorkflowID: ptr.String("alien-workflow-id"), RunID: ptr.String("run-id-2")},
+	}
+
+	tests := []struct {
+		name                     string
+		filterFn                 filterFnForVisibility
+		expectedFilteredTaskList []*indexer.Message
+	}{
+		{"empty filter always return true", buildFilterFnForVisibility("", ""), allMessages},
+		{
+			"filter with workflowId only",
+			buildFilterFnForVisibility("my-workflow-id", ""),
+			[]*indexer.Message{
+				{WorkflowID: ptr.String("my-workflow-id")},
+				{WorkflowID: ptr.String("my-workflow-id"), RunID: ptr.String("run-id-1")},
+				{WorkflowID: ptr.String("my-workflow-id"), RunID: ptr.String("run-id-2")},
+			},
+		},
+		{
+			"filter with runId only",
+			buildFilterFnForVisibility("", "run-id-1"),
+			[]*indexer.Message{
+				{RunID: ptr.String("run-id-1")},
+				{WorkflowID: ptr.String("my-workflow-id"), RunID: ptr.String("run-id-1")},
+				{WorkflowID: ptr.String("alien-workflow-id"), RunID: ptr.String("run-id-1")},
+			},
+		},
+		{
+			"filter with workflow and runId",
+			buildFilterFnForVisibility("my-workflow-id", "run-id-1"),
+			[]*indexer.Message{
+				{WorkflowID: ptr.String("my-workflow-id"), RunID: ptr.String("run-id-1")},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var result []*indexer.Message
+
+			for _, task := range allMessages {
+				if tt.filterFn(task) {
+					result = append(result, task)
+				}
+			}
+
+			assert.Equal(t, tt.expectedFilteredTaskList, result)
+		})
+	}
+}
+
+func TestGetOutputFile(t *testing.T) {
+	t.Run("returns stdout if filename is empty", func(t *testing.T) {
+		file, err := getOutputFile("")
+		assert.NoError(t, err)
+		assert.Equal(t, os.Stdout, file)
+	})
+
+	t.Run("creates a file", func(t *testing.T) {
+		file, err := getOutputFile("test.txt")
```

Comment:
nit there is a createTempFileWithContent function in the repo that might be useful